File: backend/app/api/schemas/skill.py
"""Public HTTP response schema for skills."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class SkillResponse(BaseModel):
    """Serialized public skill returned by the API."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    category: str
    sort_order: int
    created_at: datetime
    updated_at: datetime

File: frontend/src/schemas/skill.ts
import { z } from "zod";

export const skillSchema = z.object({
  id: z.uuid(),
  name: z.string(),
  category: z.string(),
  sort_order: z.number().int(),
  created_at: z.iso.datetime(),
  updated_at: z.iso.datetime(),
});

export const skillsSchema = z.array(skillSchema);
